### PR TITLE
Update to latest substrate master

### PR DESCRIPTION
**Pull request by NikVolf**
- composite accounts
- exists -> contains_key
- spawn futures
- optimised block construction
- rpc extensions update

### CODE REVIEW

File: runtime/common/src/registrar.rs [hunk 1]
```diff
@@ -703,22 +702,21 @@ mod tests {
...
-		type OnReapAccount = System;
-		type Balance = Balance;
-		type Event = ();
+		type Balance = u128;
 		type DustRemoval = ();
+		type Event = ();
 		type ExistentialDeposit = ExistentialDeposit;
-		type TransferPayment = ();
-		type CreationFee = CreationFee;
+		type AccountStore = System;
```
Comment (andresilva) [inline runtime/common/src/registrar.rs]: Why remove this?

File: service/src/chain_spec.rs [hunk 2]
```diff
@@ -171,9 +171,7 @@ fn staging_testnet_config_genesis() -> polkadot::GenesisConfig {
 				.collect(),
 		}),
 		indices: Some(polkadot::IndicesConfig {
-			ids: endowed_accounts.iter().cloned()
-				.chain(initial_authorities.iter().map(|x| x.0.clone()))
-				.collect::<Vec<_>>(),
+			indices: vec![],
```
Comment (andresilva) [inline service/src/chain_spec.rs]: Why remove this?

File: service/src/chain_spec.rs [hunk 3]
```diff
@@ -313,7 +311,7 @@ pub fn testnet_genesis(
 			changes_trie_config: Default::default(),
 		}),
 		indices: Some(polkadot::IndicesConfig {
-			ids: endowed_accounts.clone(),
+			indices: vec![],
```
Comment (andresilva) [inline service/src/chain_spec.rs]: Same as above.

File: service/src/chain_spec.rs (hunk 2 above)
Comment (shawntabrizi) [reply service/src/chain_spec.rs]: Indicies are now an "opt in" thing.

So by default, accounts won't have them.

File: service/src/chain_spec.rs (hunk 3 above)
Comment (NikVolf) [reply service/src/chain_spec.rs]: see @shawntabrizi answer

**Comment by NikVolf** [comment]
> changes lgtm. should we close #834?

@gavofyork ^^^ ?

**Pull request merged**